src/Chapter4.hs

```diff
@@ -293,7 +296,13 @@ values and apply them to the type level?
 -}
 instance Functor (Secret e) where
     fmap :: (a -> b) -> Secret e a -> Secret e b
-    fmap = error "fmap for Box: not implemented!"
+    fmap f r = case r of
+      Reward a -> Reward (f a)
+      Trap e   -> Trap e
+-- Question for reviewers: why does the linter complain about unexaustive pattern in guard version below?
+    -- fmap
+    --   | f (Reward a) = Reward (f a)
+    --   | _ (Trap e)   = Trap e
```

Comment:
I don't think that this compile.
Guards check some conditions, but here you try to pattern match on two different arguments 🤔 
What linter do you use, that warns that? Maybe it thinks that `f (Reward a)` is a boolean, so it could not tell that the second case with the type hole is covering everything. I am not sure, but the code wouldn't compile, I guess 🙂  Hope this helps!

Reply:
Thanks!